CHANGES.md

```diff
@@ -1,6 +1,7 @@
 Airbase 117
 
 * Add `air.test.fail-if-no-tests` property
+* Remove test providers from Surefire plugin
```

Comment:
"Fix build failure when there are Junit tests and no TestNG tests"
also add a note about a backward incompatible change

> However, if a
module uses both, only the JUnit tests will run -- the TestNG tests
will be silently skipped.

this is gonna be a problem for Trino, as there is a junit test there already.
With this change, we will silently looe test coverage.
Is there any way to either keep test providers but avoid failure, or have a check for mixed junit+testng modules?
Maybe we could have our own version of these providers?

The failure is:
```
java.lang.NoClassDefFoundError: org/testng/TestNG
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:87)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:145)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.lang.ClassNotFoundException: org.testng.TestNG
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 8 more
```

so maybe it would be sufficient to have an activator that triggers on "has testng on classpath" condition.

Reply:
Isn’t that as simple as configuring the surefire plugin with the necessary providers directly in Trino when we update to this version of airbase?

Reply:
Yes, any modules with mixed tests need to declare both. If we consider mixed to be “transitional”, with a goal of migrating fully from one to the other, then this seems reasonable.

Maybe @jvanzyl could get upstream Surefire fixed to work automatically with modules having both TestNG and JUnit 5?

Reply:
> Isn’t that as simple as configuring the surefire plugin with the necessary providers directly in Trino

That might do, if we assume Trino is the only downstream consumer of the project.

Reply:
It’s not, but if this is documented as a behavior change, it should be a hint for users to update their project accordingly.